st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -445,19 +481,15 @@ def delete(self, exec_id):
 
         try:
             (liveaction_db, execution_db) = action_service.request_cancellation(
-                liveaction_db, get_requester())
+                liveaction_db, requester_user.name or cfg.CONF.system_user.user)
```

Comment:
this utility function expected to magically get `pecan.request` global ten levels into the call stack. This way you can clearly see where this requester comes from.